Pocket/pocket-ios
# Add Highlights

## Summary
* This PR adds the ability to create highlights from iOS

## References 
*  branch name
## Implementation Details
* Update `SavedItemViewModel`, add logic to:
    - Create a patch from a quote in a component
    - Create and save a local `Highlight`
    - Sync the highlight to the backend
- Update `ArticleComponentTextCell`, add logic to capture an highlightable range and send it to the view model via callback

## Test Steps
Premium users:
* Open an article in Reder, select some text:
    - if the selection includes an highlight, make sure you don't see an Highlight button in the context menu
    - if the selection does not include an highlight, make sure you see the Highlight button
    - Tap "Highlight" and make sure the highlight gets added
    - Optional: check on another platform that the highlight exists
Free users:
- Same as above but if you have already saved 3 highlights, the "Highlight" button should do nothing

> [!NOTE] 
> the alert/premium upsell will be added in another PR
 

## PR Checklist:

## Screenshots
